2002-04-14  Grant Edwards <grante@visi.com>

        * stage2/builtins.c (terminal_func):  Force VGA16 code
        to use "text" display if serial console is selected.
        Enables use of serial and splashimage/vga16 console
        combination.
diff -U5 -r grub-0.91-vga/stage2/builtins.c grub-0.91-vga-fixed/stage2/builtins.c
--- grub-0.91-vga/stage2/builtins.c	Sun Apr 14 13:18:30 2002
+++ grub-0.91-vga-fixed/stage2/builtins.c	Sun Apr 14 10:51:53 2002
@@ -159,10 +159,12 @@
   BUILTIN_CMDLINE | BUILTIN_MENU,
   "background RRGGBB",
   "Sets the background color when in graphics mode."
   "RR is red, GG is green, and BB blue. Numbers must be in hexadecimal."
 };
+
+extern int vga_disabled;
 #endif  /* VGA16 */
 
 
 /* blocklist */
 static int
@@ -4170,11 +4172,15 @@
 	    }
 	  else if ((terminal & TERMINAL_SERIAL) && serial_checkkey () != -1)
 	    {
 	      terminal &= (TERMINAL_SERIAL | TERMINAL_DUMB);
 	      (void) getkey ();
-
+#ifdef VGA16
+              vga_disabled = 1;
+              display_idx = 0;
+              display = &display_entries[0];
+#endif
 	      /* If the interface is currently the command-line, restart
 		 it to repaint the screen.  */
 	      if (flags & BUILTIN_CMDLINE)
 		grub_longjmp (restart_cmdline_env, 0);
 	      
@@ -4182,10 +4188,18 @@
 	    }
 	}
 
       /* Expired.  */
       terminal &= (default_terminal | TERMINAL_DUMB);
+#ifdef VGA16
+      if (!(terminal & TERMINAL_CONSOLE))
+        {
+          vga_disabled = 1;
+          display_idx = 0;
+          display = &display_entries[0];
+        }
+#endif
     }
 #endif /* SUPPORT_SERIAL */
 
   return 0;
 }
